packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/link/read.rs

```diff
@@ -3,118 +3,163 @@ use error_stack::{IntoReport, Report, Result, ResultExt};
 use tokio_postgres::GenericClient;
 
 use crate::{
-    knowledge::{EntityId, Link, Links, Outgoing},
-    ontology::types::uri::VersionedUri,
-    store::{crud, AsClient, PostgresStore, QueryError},
+    knowledge::{EntityId, Links, Outgoing},
+    ontology::types::uri::{BaseUri, VersionedUri},
+    store::{
+        crud,
+        query::{LinkQuery, OntologyVersion},
+        AsClient, PostgresStore, QueryError,
+    },
 };
 
-// TODO: we should probably support taking PersistedEntityIdentifier here as well as an EntityId
-#[async_trait]
-impl<C: AsClient> crud::Read<'_, EntityId, Link> for PostgresStore<C> {
-    type Output = Links;
-
-    async fn get(&self, identifier: EntityId) -> Result<Self::Output, QueryError> {
-        let multi_links = self
-            .client
-            .as_client()
-            .query(
-                r#"
-                WITH aggregated as (
-                    SELECT link_type_version_id, ARRAY_AGG(target_entity_id ORDER BY multi_order ASC) as links
-                    FROM links
-                    WHERE active AND multi and source_entity_id = $1
-                    GROUP BY link_type_version_id
-                )
-                SELECT base_uri, "version", links FROM aggregated
-                INNER JOIN ids ON ids.version_id = aggregated.link_type_version_id
-                "#,
-                &[&identifier],
+async fn single_by_source_entity_id(
+    client: &(impl GenericClient + Sync),
+    source_entity_id: EntityId,
+) -> Result<impl Iterator<Item = (VersionedUri, Outgoing)> + Send, QueryError> {
+    Ok(client
+        .query(
+            r#"
+            SELECT base_uri, "version", target_entity_id
+            FROM links
+            INNER JOIN ids ON ids.version_id = links.link_type_version_id
+            WHERE active AND NOT multi and source_entity_id = $1
+            "#,
+            &[&source_entity_id],
```

Comment:
Here `parameter_list` can be utilized as well

Reply:
Not possible here because it is using `query` rather than `query_raw`